### PR TITLE
Mobile tests

### PR DESCRIPTION
**Pull request by paweldomas**

### CODE REVIEW

File: src/test/java/org/jitsi/meet/test/base/FailureListener.java [hunk 1]
```diff
@@ -340,24 +261,20 @@ private void saveThreadDump(String fileNamePrefix)
...
-                new File(outputLogsParentFolder,
-                    fileNamePrefix + suffix + "-driver" + extension)));
+            BufferedWriter out
+                = new BufferedWriter(
+                    new FileWriter(
+                        new File(
+                            outputLogsParentFolder,
+                            fileNamePrefix + suffix + "-driver" + extension)));
 
             Iterator<LogEntry> iter = logs.iterator();
```
[the remark this answers is not in the document]
Comment (damencho) [reply src/test/java/org/jitsi/meet/test/base/FailureListener.java]: because of logs been null.

**Comment by paweldomas** [comment]
@damencho I think it's now ready for review. I've fixed the NPE issue

**Comment by damencho** [comment]
Thanks, I will look at the PR first thing today. I already had a quick look at all commits and it looks ok.

**Comment by paweldomas** [comment]
Thanks !

**Comment by damencho** [comment]
@paweldomas There was a failure in the PSNR tests. It seems that the screenshots folder is not created:
java.lang.RuntimeException: Failed to rename screenshot file to directory name: target/surefire-reports/screenshots, file name: org.jitsi.meet.test.PSNRTest-participant1.png, original file: /tmp/screenshot6796829592652330920.png
	at org.jitsi.meet.test.base.Participant.takeScreenshot(Participant.java:391)
	at org.jitsi.meet.test.base.FailureListener.lambda$takeScreenshots$0(FailureListener.java:186)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at org.jitsi.meet.test.base.FailureListener.takeScreenshots(FailureListener.java:185)

**Comment by paweldomas** [comment]
oh why didn't this test fail during PR tests ?

**Comment by paweldomas** [comment]
oh I see it's only on failure. I will take a look

**Comment by paweldomas** [comment]
from the code it looks that "mkdirs" is called on the screenshots folder. Could it be a permissions problem ?

**Comment by damencho** [comment]
No, I don't think so. This is a task which had been running for long time and had failures before. But I will check it to make sure.

**Comment by paweldomas** [comment]
It looks like the screenshots subdirectory is not created

**Comment by paweldomas** [comment]
btw this is not the root cause of the failure

**Comment by damencho** [comment]
Nope, its just an error after test failure.

**Comment by paweldomas** [comment]
So the failure was very random I would say. Have we ever seen PSNT test to fail ? PR tests are writing screenshots fine, aren't they ? The difference is that PRs do not use "target/surefire-reports" output dir.

**Comment by damencho** [comment]
I haven't check the pr tests. There is a property to set the name of the report folder to be able to have different for different browser combinations. But this property can be omitted and then default one is used. Screenshot's folder need to be created in all cases.

**Comment by paweldomas** [comment]
Right. As far as the code goes it should be. I'm trying to figure out why this failed.

**Pull request merged**